src/finalisers/esm.ts

```diff
@@ -82,6 +82,11 @@ export default function esm(
 
 	const exportBlock: string[] = [];
 	const exportDeclaration: string[] = [];
+	exports.forEach(specifier => {
+		if (specifier.shim) {
+			exportBlock.push(`const ${specifier.local} = null;`);
```

Comment:
This can be done in the `preRender` of `Chunk.ts`.

Reply:
Also the `const` should be a `${varOrConst}` reference.

Reply:
Just like at https://github.com/rollup/rollup/blob/79ad616596f8c5cebafa62883f4ddd63d922bbb5/src/Chunk.ts#L502, when needing a shim variable you can add it to the deshadowing list this way.

preRender renders the main body source, while render renders the finalizers. I think preRender would be a good place. Note that there is a `hoistedSource` for bindings that must be available early this could be added to at https://github.com/rollup/rollup/blob/79ad616596f8c5cebafa62883f4ddd63d922bbb5/src/Chunk.ts#L778. Just writing directly into that string is fine (if (shimExports) hoistedSource += `${varOrConst} $$shim = null`;). Note that finalizers will apply wrappers so that there are no global leaks to worry about.

Reply:
If I do this, it works for everything except systemjs, where the output is

```js
System.register([], function (exports, module) {
	'use strict';
	return {
		execute: function () {

			exports('$$shim$2', $$shim$2);
			var $$shim$2 = null;

			const $$shim = exports('$$shim$1', null);

		}
	};
});
```

I may have the change the order of `functionExports` in systemjs to get it to work this way.

Reply:
made the change.